src/header/js/header-menus.js

```diff
@@ -307,6 +377,148 @@ export default ({variant = 'desktop'}) => {
+		const categoryHeader = category.target.parentNode.querySelector('.esri-header-menus-flyout--categories-item_header');
+		const active = categoryList.getAttribute('aria-expanded') === 'false' ? 'true' : 'false';
+		const categoryDetailsItems = [].slice.call(category.target.parentNode.querySelectorAll('.esri-header-menus-flyout--categories-details_item'));
+		const catsComputedStyle = window.getComputedStyle(categoryDetailsItems[0]);
+		const computedHeight = (parseInt(catsComputedStyle.height) * categoryDetailsItems.length);
+		const computedMargin = (parseInt(catsComputedStyle.marginTop) * categoryDetailsItems.length) + parseInt(catsComputedStyle.marginTop);
+		const headers = [].slice.call(document.querySelectorAll('.esri-header-menus-flyout--categories-item_header'));
+		const items = [].slice.call(document.querySelectorAll('.esri-header-menus-flyout--categories-item'));
+		const itemsList = [].slice.call(document.querySelectorAll('.esri-header-menus-flyout--list-items'));
+		const isMobile = (window.innerWidth < 1024);
+
+		if (isMobile) {
+			const categoryListArr = [].slice.call(document.querySelectorAll('.esri-header-menus-flyout--categories-details[aria-expanded]'));
+			categoryListArr.forEach((list) => {
+				list.setAttribute('aria-expanded', 'false');
+				list.style.height = '0';
+			});
+	
+			categoryList.setAttribute('aria-expanded', `${active}`);
+			if (active === 'true') {
+				categoryList.style.height = `${(computedHeight) + (computedMargin)}px`;
+				headers.forEach((head) => {
+					head.setAttribute('aria-current', 'false');
+				});
+				categoryHeader.setAttribute('aria-current', 'true');
+			} else {
+				categoryList.style.height = '0';
+				categoryHeader.setAttribute('aria-current', 'false');
+			}
+		} else {
+			items.forEach((item, index) => {
+				item.addEventListener('click', (e) => {
+					const parentNode = e.target.parentNode;
+					const selectedCategory = parentNode.hasAttribute('data-id') && parentNode.getAttribute('data-id');
+					const selectedList = itemsList[index].hasAttribute('data-id') && itemsList[index].getAttribute('data-id');
+					const selectedListCols = itemsList[index].hasAttribute('data-coltype') && itemsList[index].getAttribute('data-coltype');
+					
+					itemsList.forEach((list, index) => {
+						list.setAttribute('aria-current', 'false');
+						items[index].setAttribute('aria-current', 'false');
+					});
+
+					if (selectedCategory === selectedList) {
+						(selectedListCols) === '1' ? resetFlyoutDimensions(parentNode) : resetFlyoutDimensions('disabled');
+						parentNode.setAttribute('aria-current', 'true');
+						itemsList[index].setAttribute('aria-current', 'true');
+					}
+				});
+			});
+		}
+	}
+	
+	function renderFlyoutMenu(items, type, id, uuid, suuid) {
+		const $items = [];
+		const listArr = [];
+		let category = "";
+
+		if (type === 'category') {
```

Comment:
this should be a switch statement - not chained if statements since you evaluating the same variable

Reply:
the check of items.cols && items.cols.length is also repeated that should only need to be checked once

Reply:
@paulcpederson I think this is the final updates.. Can I please get your review on this?